backend/models/time.py

```diff
@@ -28,27 +28,61 @@ def _retry_if_result_none(result):
 
 
 # TODO - Abstract as APIMeasurer.
-# TODO - Better handling of api_url and api_key.
-class APIDrivingTime(Measurer):
+# TODO - Better handling of api_url and access_token.
+class APITime(Measurer):
     """Class for API driving time measurements."""
 
-    def __init__(self, api_key=None, early_exit_outer_radius=120.0 * 10**3, *args, **kwargs):
+    def __init__(
+            self, access_token='', api_url='',
+            early_exit_outer_radius=120.0 * 10**3,
+            *args, **kwargs):
         """Initialize the distance class with for an API measurer."""
-        self.api_key = api_key
+        self.access_token = access_token
+        self.api_url = api_url
         self._haversine_measurer = distance.HaversineDistance()
         self.early_exit_outer_radius = early_exit_outer_radius
         # Some APIs only accept a maximum number of entries per call.
         self.max_matrix_size = None
 
-    @retry(wait_fixed=500, stop_max_attempt_number=2)
+    @retry(retry_on_result=_retry_if_result_none, stop_max_attempt_number=10, wait_fixed=2000)
     def _get_matrix(self, source_points, destination_points):
```

Comment:
IMO this should still raise a not implemented error and there can be another class that the Mapbox / OSRM classes inherit from

Reply:
This is ultimately very specific: it's expecting points to be in a certain format and for `durations` to be a key in the API response. Seems like that's a bit too specific for the base class to know.

Reply:
self, source_points, destination_points are in the same format and the output is the same so idk. I'll think about it

Reply:
doc string seems specific at least. I kinda agree with brian on this one